Kernel/Arch/DeferredCallPool.h

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2020, Tom <tomut@yahoo.com>
+ * Copyright (c) 2023, Timon Kruiper <timonkruiper@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <Kernel/Arch/DeferredCallEntry.h>
+
+namespace Kernel {
+
+class DeferredCallPool {
+public:
+    void init();
+    void execute_pending();
+    DeferredCallEntry* get_free();
+    void return_to_pool(DeferredCallEntry*);
+    void queue_entry(DeferredCallEntry*);
+
+private:
+    DeferredCallEntry* m_pending_deferred_calls; // in reverse order
```

Comment:
(~~comment is somewhat helpful but itʼd be better if you indicate what the reverse order is related to exactly—maybe using different phrasing like “registration order” or “execution order” could be the thing also? 🤔~~)

seems to be copied, right?

Reply:
Idem ditto.